pkg/solver/che_routing.go

```diff
@@ -50,6 +50,14 @@ var (
 	configMapDiffOpts = cmpopts.IgnoreFields(corev1.ConfigMap{}, "TypeMeta", "ObjectMeta")
 )
 
+// keys are port numbers, values are maps where keys are endpoint names (in case we need more than 1 endpoint for a single port) and values
+// contain info about the intended endpoint scheme and the order in which the port is defined (used for unique naming)
+type portMapping map[int32]map[string]portMappingValue
```

Comment:
why `int32` and not simple `int` ?

Reply:
This is compared with `ServicePort.Port` which is an `int32` in `che_routing.go:findServiceForPort()`. I wanted to keep the types that are used in the k8s API.